### DISCUSSION TITLE
Interrupt SASSession and clean up properly

### DISCUSSION THREAD

**Discussion by nbuck1234** [sassoftware/saspy#591] [Q&A]
I am using winiom configuration , from a remote machine to our sas server.  I happen to be using this with dagster if you're curious. 

My scenario is in the case where the job is kicked off, I want to be able to cancel it from the remote machine, rather than tracking it down in the SAS server. 

My code looks like this : 

```
        sas= saspy.SASsession(cfgname= 'winiomwin')
        try:
            #sas.set_batch(True)
            context.log.info("The SAS Process ID is : "  + str(sas.SASpid))
            j=sas.submit('%include "' + program_path + '";')         
            with open(logfile_path, 'w') as f:
                f.write(j["LOG"])
            sas._endsas()
            log= j["LOG"]
        except DagsterExecutionInterruptedError as e:
            sas._endsas()
            
            raise e
```

When I kill the running job I get the following in my stdout, which is what I expect:

```
SAS Connection established. Subprocess id is 21244

SAS didn't shutdown w/in 5 seconds; killing it to be sure
SAS Connection terminated. Subprocess id was 21244

```

In this specific test case above i got the SAS process id from the session object : 

> The SAS Process ID is : 31084

When I logged into the SAS server to see if it was really ended, I see that it is in fact still running, and I must manually end it from there.

Whats the difference between the subprocess id from the stdout and the SAS process id property of the sassession?
How can i be sure that the process is cleaned up properly in this scenario?

I hope this question is useful to the community!

**Comment by tomweber-sas** [comment]
Hey, yes, I can help with this. Actually, this would be an enhancement request, which would be good to be an Issue. Do you mind changing it to an Issue, or opening it as one? There's a choice to 'reference in new issue' but I think if I click that I'll be the one opening the issue.

The IOM access method uses Java since there's no Python IOM client, so I use the Java IOM client and have an interface between Python and Java to do that. So, the subprocess in this case is the Java process. IOM is an asynchronous interface, so it seems like I may need to be doing something extra that I'm not, to terminate the SAS side, when you terminate the Python side abnormally. I have an idea but need to investigate.

Thanks,
Tom

**Reply by nbuck1234** [reply]
Thanks so much!
I will open this as an issue.

Would you like me to send you contact info?
